src/concern/mod.rs

```diff
@@ -276,4 +279,34 @@ impl WriteConcern {
 
         Ok(())
     }
+
+    pub(crate) fn serialize_for_client_options<S>(
```

Comment:
Ditto above comment: instead of having this extra function we can just define the `WriteConcernHelper` function directly in the `Serialize` implementation.

Reply:
This one I am a little more hesistant about per my above comment - I feel like there's a little more going on under the hood with this serialize (i.e. two renames, and a `serialize_with`) and I'm not sure I like the idea of making `ClientOptionsHelper` even less readable.

Of course, this custom serializer is only being used by client options, so maybe it does make more sense to keep everything in one place. I think we should probably touch base on this in an actual conversation.

Reply:
I'm using helper structs with `Serialize` derived more consistently across these functions now. Check that out in the most recent commit and let me know what you think.